fe/fe-core/src/main/java/org/apache/doris/nereids/glue/translator/PhysicalPlanTranslator.java

```diff
@@ -318,7 +320,8 @@ public PlanFragment visitPhysicalDistribute(PhysicalDistribute<? extends Plan> d
                 dataStreamSink.setOutputTupleDesc(projectionTuple);
             }
         }
-        DataPartition dataPartition = toDataPartition(distribute.getDistributionSpec(), validOutputIds, context);
+        DataPartition dataPartition = toDataPartition(distribute.getDistributionSpec(), validOutputIds, context,
```

Comment:
do u test, bucket shuffle join with left table is olap table and right table is a bucketed hive table?
i think the result will be wrong because right table will be shuffled by spark hash and left table is distributed by Doris hash.
i think the right way to do this in Nereids should add a new ShuffleType in DistributionSpecHash, and generate PhysicalDistribute with **RIGHT** ShuffleType when adding enforcer

Reply:
yes, makes sense. I'll add it